feat(types,clerk-js): Introduce Custom OAuth providers (#3678)
clerk/javascript: .changeset/unlucky-hornets-roll.md

---

> "@clerk/clerk-js": patch

Reviewer: ❓  Should we go for `minor` here? Isn't this a new feature?

Author: Correct @anagstef, I will update the changeset 🙏

Author: Done

"@clerk/types": patch
---

Added support for Custom OAuth providers

> - Updated strategy types to include CustomOAuthStrategy:

Reviewer 2: ```suggestion
- Updated strategy types to include `CustomOAuthStrategy`:
```

>   - Added `export type CustomOAuthStrategy = `oauth_custom_${string}`;`

Reviewer: ```suggestion
  - Added the `CustomOAuthStrategy` type with the value `oauth_custom_${string}`
```

Author: Done

>   - Modified `OAuthStrategy` to include CustomOAuthStrategy:

Reviewer 2: ```suggestion
  - Modified `OAuthStrategy` to include `CustomOAuthStrategy`:
```

    `export type OAuthStrategy = `oauth_${OAuthProvider}` | CustomOAuthStrategy;`

> - Added new type: `export type CustomOauthProvider = `custom_${string}`;` and extend `OAuthProvider` type to include `CustomOauthProvider`

Reviewer: ```suggestion
- Added the `CustomOauthProvider` type and extended `OAuthProvider` type to include `CustomOauthProvider`
```

Author: Done

> - Added support for displaying provider initials when logo_url is null for custom OAuth providers

Reviewer: ```suggestion
- Added support for displaying provider initials when `logo_url` is null for custom OAuth providers
```

Author: Done

> - Created new `InitialIcon` component in order to display custom oauth provider initials if provider `logo_url` is null

Reviewer 3: ```suggestion
- Created new `ProviderInitialIcon` internal component in order to display custom oauth provider initials if provider `logo_url` is null
```
